core/src/integration-test/java/io/aiven/kafka/tieredstorage/RemoteStorageManagerTest.java

```diff
@@ -663,4 +673,74 @@ private void writeManifest(final ObjectKeyFactory objectKeyFactory) throws IOExc
         Files.createDirectories(manifestPath.getParent());
         Files.writeString(manifestPath, manifest);
     }
+
+    @ParameterizedTest
+    @MethodSource("provideInterruptionExceptions")
+    void fetchSegmentInterruptionWhenGettingManifest(final Class<Exception> outerExceptionClass,
+                                                     final Class<Exception> exceptionClass) throws Exception {
+        final SegmentManifestProvider segmentManifestProvider = mock(SegmentManifestProvider.class);
+        when(segmentManifestProvider.get(any())).thenAnswer(invocation -> {
+            final Exception innerException = exceptionClass.getDeclaredConstructor().newInstance();
+            if (outerExceptionClass != null) {
+                throw outerExceptionClass.getDeclaredConstructor(String.class, Throwable.class)
+                    .newInstance("", innerException);
+            } else {
+                throw innerException;
+            }
+        });
+
+        final var config = Map.of(
+            "chunk.size", "1",
+            "storage.backend.class", "io.aiven.kafka.tieredstorage.storage.filesystem.FileSystemStorage",
+            "storage.root", targetDir.toString()
+        );
+        rsm.configure(config);
+        rsm.setSegmentManifestProvider(segmentManifestProvider);
+
+        final InputStream inputStream = rsm.fetchLogSegment(REMOTE_LOG_METADATA, 0);
+        assertThat(inputStream.readAllBytes()).isEmpty();
```

Comment:
I gues just `isEmpty()` is enough, same in the other test.

Reply:
Yep, fixed